include/seqan/sequence/string_set_base.h

```diff
@@ -1598,10 +1598,10 @@ concat(StringSet<TString, TSpec> const & constMe)
 }
 
 template <typename TStrings, typename TDelim>
-inline String<typename Value<typename Value<TStrings>::Type>::Type>
+inline typename Value<TStrings>::Type
```

Comment:
this specialization is called when you concatenate with a delimeter. Maybe `Concatenator<TSTrings>`  would be better, I don't know. I also don't know why it was done in the way it was before.